src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -124,7 +126,8 @@ void GcsResourceManager::HandleGetAllAvailableResources(
   ++counts_[CountType::GET_ALL_AVAILABLE_RESOURCES_REQUEST];
 }
 
-void GcsResourceManager::UpdateFromResourceReport(const rpc::ResourcesData &data) {
+void GcsResourceManager::UpdateFromResourceReport(const rpc::ResourcesData &data,
+                                                  bool from_resource_view) {
```

Comment:
(Probably too much for this PR) Ideally instead of having a `bool from_resource_view`, we should have two completely different code path for consuming two different types of messages.

Reply:
i think it's a good idea. Let me try.